From f54b95a6334ba02e236419ca2dbd0d8294f1db0c Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Mon, 8 Nov 2021 09:50:15 +0100
Subject: [PATCH 144/179] x86: irq_pipeline: Fix condition for !IRQ_PIPELINE
 case

This code is already under CONFIG_SMP: The comment got it right, not yet
the implementation. Only affects the pipeline-off case.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 arch/x86/kernel/apic/vector.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kernel/apic/vector.c b/arch/x86/kernel/apic/vector.c
index 54e761724..1cb7ab408 100644
--- a/arch/x86/kernel/apic/vector.c
+++ b/arch/x86/kernel/apic/vector.c
@@ -758,7 +758,7 @@ static struct irq_desc *__setup_vector_irq(int vector)
 	int isairq = vector - ISA_IRQ_VECTOR(0);
 
 	/* Copy the cleanup vector if irqs are pipelined. */
-	if (IS_ENABLED(CONFIG_SMP) &&
+	if (IS_ENABLED(CONFIG_IRQ_PIPELINE) &&
 		vector == IRQ_MOVE_CLEANUP_VECTOR)
 		return irq_to_desc(IRQ_MOVE_CLEANUP_VECTOR); /* 1:1 mapping */
 	/* Check whether the irq is in the legacy space */
-- 
2.38.1

